src/Application/ApplicationFileProcessor.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\Core\Application;
+
+use Rector\ChangesReporting\Application\ErrorAndDiffCollector;
+use Rector\Core\Configuration\Configuration;
+use Rector\Core\Contract\Processor\FileProcessorInterface;
+use Rector\Core\FileSystem\FilesFinder;
+use Symplify\SmartFileSystem\SmartFileInfo;
+use Symplify\SmartFileSystem\SmartFileSystem;
+
+final class ApplicationFileProcessor
+{
+    /**
+     * @var FileProcessorInterface[]
+     */
+    private $fileProcessors = [];
+
+    /**
+     * @var SmartFileSystem
+     */
+    private $smartFileSystem;
+
+    /**
+     * @var Configuration
+     */
+    private $configuration;
+
+    /**
+     * @var ErrorAndDiffCollector
+     */
+    private $errorAndDiffCollector;
+
+    /**
+     * @var FilesFinder
+     */
+    private $filesFinder;
+
+    /**
+     * @param FileProcessorInterface[] $fileProcessors
+     */
+    public function __construct(
+        FilesFinder $filesFinder,
+        ErrorAndDiffCollector $errorAndDiffCollector,
+        Configuration $configuration,
+        SmartFileSystem $smartFileSystem,
+        array $fileProcessors = []
+    ) {
+        $this->fileProcessors = $fileProcessors;
+        $this->smartFileSystem = $smartFileSystem;
+        $this->filesFinder = $filesFinder;
+        $this->errorAndDiffCollector = $errorAndDiffCollector;
+        $this->configuration = $configuration;
+    }
+
+//    /**
+//     * @param string[] $smartFileInfo
+//     */
+    public function run(SmartFileInfo $smartFileInfo): string
+    {
+        $newContent = null;
+
+        foreach ($this->fileProcessors as $fileProcessor) {
+            if (! $fileProcessor->supports($smartFileInfo)) {
+                continue;
+            }
+
+            $newContent = $fileProcessor->process($smartFileInfo);
```

Comment:
Imho we need content here as a parameter because diff in dry run will not work when one file is changed by more processors - first change content but it is not saved, then the second read the content from unchanged file

Reply:
This is WIP, not working :) Wait for review till CI is green.

Reply:
Ok ok, I just wanted to inform you about the situation :)

Reply:
Thanks! :) That's where I got stuck yesterday.

It will probably require value object holding both content and file info reference. Like PHP_CodeSniffer has.